test/iterators.jl

```diff
@@ -321,7 +321,7 @@ end
 @test Base.IteratorSize(product(1:2))                        == Base.HasShape{1}()
 @test Base.IteratorSize(product(1:2, 1:2))                   == Base.HasShape{2}()
 @test Base.IteratorSize(product(take(1:2, 1), take(1:2, 1))) == Base.HasShape{2}()
-@test Base.IteratorSize(product(take(1:2, 2)))               == Base.HasShape{2}()
+@test Base.IteratorSize(product(take(1:2, 2)))               == Base.HasShape{1}()
```

Comment:
Should this change be moved to the first commit, or is it ok to just squash them?

Reply:
Ah, good catch. They could be squashed, but since the first one makes sense on its own, probably better keep them separate. I've moved the change to the first commit.